subsys/pm/device.c

```diff
@@ -25,12 +25,25 @@ const char *pm_device_state_str(enum pm_device_state state)
 	}
 }
 
+static int domain_supported_cb(const struct device *dev, void *context)
+{
+	struct pm_device *pm = dev->pm;
+	enum pm_device_action action = (enum pm_device_action)context;
+
+	if (pm->action_cb != NULL) {
+		(void)pm->action_cb(dev, action);
+	}
+
+	return 0;
+}
+
 int pm_device_state_set(const struct device *dev,
 			enum pm_device_state state)
```

Comment:
I think we should not update deprecated functions (`pm_device_state_set`).

Reply:
reasons ?

Reply:
fair enough